### PR TITLE
BaSP-TG-58: Super-Admins

### PR DESCRIPTION
**Pull request by lucianofrailelopez**
• GET (TG-58): tests for getAll and getById.  
• POST (TG-62): tests for creating a Super-Admins.  
• PUT (TG-65): tests for updating aSuper-Admins.  
• DELETE (TG-64): tests for deleting aSuper-Admins.

### CODE REVIEW

File: src/tests/super-admins.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+//       const data = await request(app).get('/superadmins').send();
+//       expect(data).toBeDefined();
+//     });
+  
+//     test('response should not be empty', async () => {
+//       const response = await request(app).get('/superadmins').send();
+//       expect(response.body.data.length).toBeGreaterThan(0);
+//     });
+// });
+
```
Comment (ArianaMaldonado) [inline src/tests/super-admins.test.js:34]: There's no test for getById function

File: src/tests/super-admins.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// describe('POST /superadmins', () => {
+//     test('Shold create an Super-Admin', async () => {
```
Comment (ArianaMaldonado) [inline src/tests/super-admins.test.js:36]: Be careful with the string inside `test()`, you wrote "shold" instead of "should".

**Review by ArianaMaldonado — CHANGES_REQUESTED** [review]
There is no test for getById function.

File: src/tests/super-admins.test.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+describe('GET /superadmins/:id', () => {
+  test('response should return a 200 status', async () => {
+    const response = await request(app).get(`/superadmins/${sAdmin}`).send();
+    expect(response.status).toBe(200);
+  });
+
+  test('response should return a true error if not id', async () => {
+    const response = await request(app).get('/superadmins/').send();
+    expect(response.status).toBeTruthy();
```
Comment (dayanagm96) [inline src/tests/super-admins.test.js:74]: If you are looking for a true error, the response should be 'response.error', no?

File: src/tests/super-admins.test.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  test('response should not be empty', async () => {
+    const response = await request(app).get('/superadmins').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+  });
+});
+
+describe('POST /superadmins', () => {
+  test('Should create an Super-Admin', async () => {
```
Comment (dayanagm96) [inline src/tests/super-admins.test.js:38]: The description should be 'Should create _a_ Super-Admin'. Line 79 is the same.

**Review by dayanagm96 — APPROVED** [review]

File: src/controllers/super-admins.js [hunk 5]
```diff
@@ -82,10 +82,7 @@ const createSuperAdmin = async (req, res) => {
       error: false,
     });
   } catch (error) {
-    return res.status(500).json({
-      msg: 'An error has occurred',
-      error: error.details[0].message,
-    });
+    return res.status(400);
```
Comment (lucasrendo) [inline src/controllers/super-admins.js:85]: catch error should be 500 and contain te structure { msg, data, error}

File: src/controllers/super-admins.js [hunk 6]
```diff
@@ -112,10 +109,7 @@ const updateSuperAdmin = async (req, res) => {
       error: false,
     });
   } catch (error) {
-    return res.status(500).json({
-      msg: 'There was an error',
-      error: error.details[0].message,
-    });
+    return res.status(500);
```
Comment (lucasrendo) [inline src/controllers/super-admins.js:112]: response should contain json with { msg, data, error}

File: src/seeds/tasks-seeds.js [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
-  workedHours: '20',
-  date: '2021-12-05T03:00:00.000+00:00',
-}];
+// // Generate ObjectId on https://observablehq.com/@hugodf/mongodb-objectid-generator
+// export default [{
+//   _id: mongoose.Types.ObjectId('62891835d4e286802a02756e'),
+//   decription: 'task decription',
+//   workedHours: '20',
+//   date: '2021-12-05T03:00:00.000+00:00',
+// }];
```
Comment (lucasrendo) [inline src/seeds/tasks-seeds.js:9]: this should not be commented

File: src/tests/super-admins.test.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      name: 'pab_lo',
+    });
+    expect(response.status).toBe(400);
+  });
+
+  test('if the last name is spelled wrong, it returns me an undefined data', async () => {
+    const response = await request(app).put(`/superadmins/${sAdmin}`).send({
+      lastName: 'pere*zz',
+    });
+    expect(response.status).toBe(400);
```
Comment (lucasrendo) [inline src/tests/super-admins.test.js:97]: should check for undefined data as described on the test description

File: src/tests/super-admins.test.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  test('response should return a true error if not id', async () => {
+    const response = await request(app).put('/superadmins/').send();
+    expect(response.status).toBe(404);
```
Comment (lucasrendo) [inline src/tests/super-admins.test.js:102]: should check for true error as for what the test says

**Review by lucasrendo — APPROVED** [review]

**Pull request merged**